appscale/tools/local_state.py

```diff
@@ -372,62 +357,60 @@ def update_local_metadata(cls, options, node_layout, host, instance_id):
-      'instance_id' : str(instance_id),
-      'table' : options.table,
-      'secret' : cls.get_secret_key(options.keyname),
-      'db_master' : node_layout.db_master().public_ip,
-      'ips' : all_ips,
+    appscalefile_contents = {
       'infrastructure' : infrastructure,
       'group' : options.group,
     }
 
     if infrastructure != "xen":
-      yaml_contents['zone'] = options.zone
+      appscalefile_contents['zone'] = options.zone
 
     if infrastructure == "gce":
-      yaml_contents['project'] = options.project
+      appscalefile_contents['project'] = options.project
 
     if infrastructure == 'azure':
-      yaml_contents['azure_subscription_id'] = options.azure_subscription_id
-      yaml_contents['azure_app_id'] = options.azure_app_id
-      yaml_contents['azure_app_secret_key'] = options.azure_app_secret_key
-      yaml_contents['azure_tenant_id'] = options.azure_tenant_id
-      yaml_contents['azure_resource_group'] = options.azure_resource_group
-      yaml_contents['azure_storage_account'] = options.azure_storage_account
-      yaml_contents['azure_group_tag'] = options.azure_group_tag
-
-    with open(cls.get_locations_yaml_location(options.keyname), 'w') \
-        as file_handle:
-      file_handle.write(yaml.dump(yaml_contents, default_flow_style=False))
+      appscalefile_contents['azure_subscription_id'] = options.azure_subscription_id
+      appscalefile_contents['azure_app_id'] = options.azure_app_id
+      appscalefile_contents['azure_app_secret_key'] = options.azure_app_secret_key
+      appscalefile_contents['azure_tenant_id'] = options.azure_tenant_id
+      appscalefile_contents['azure_resource_group'] = options.azure_resource_group
+      appscalefile_contents['azure_storage_account'] = options.azure_storage_account
+      appscalefile_contents['azure_group_tag'] = options.azure_group_tag
+
+    locations_json = {
+      'node_info': role_info,
+      'infrastructure_info': appscalefile_contents
+    }
 
-    # and now we can write the json metadata file
     with open(cls.get_locations_json_location(options.keyname), 'w') \
         as file_handle:
-      file_handle.write(json.dumps(role_info))
+      file_handle.write(json.dumps(locations_json))
 
 
   @classmethod
-  def get_from_yaml(cls, keyname, tag):
+  def get_from_appscalefile(cls, tag, keyname):
     """Reads the YAML-encoded metadata on disk and returns the value associated
     with the given tag.
 
     Args:
       keyname: A str that indicates the name of the SSH keypair that
         uniquely identifies this AppScale deployment.
-      tag: A str that indicates what we should look for in the YAML file.
+      tag: A str that indicates what we should look for in the AppScalefile.
     """
-    with open(cls.get_locations_yaml_location(keyname), 'r') as file_handle:
-      locations_yaml = yaml.safe_load(file_handle.read())
-      return locations_yaml[tag]
+    if not os.path.exists(cls.get_locations_json_location(keyname)):
```

Comment:
You certainly can if you want. There are probably lots of existing places in our codebase where we check first before opening, but I think it would be out of scope for this pull request to change all of them.